bigtable-hbase-2.x-parent/bigtable-hbase-2.x/src/main/java/com/google/cloud/bigtable/hbase2_x/BigtableAdmin.java

```diff
@@ -213,76 +258,137 @@ public void snapshotAsync(SnapshotDescription snapshot)
   }
 
   @Override
-  public Future<Void> addColumnFamilyAsync(TableName arg0, ColumnFamilyDescriptor arg1)
+  public Future<Void> addColumnFamilyAsync(TableName tableName, ColumnFamilyDescriptor columnFamily)
       throws IOException {
-    throw new UnsupportedOperationException("addColumnFamilyAsync"); 
+    String columnName = columnFamily.getNameAsString();
+    Modification.Builder modification = Modification.newBuilder().setId(columnName)
+        .setCreate(new ColumnDescriptorAdapter()
+        .adapt((HColumnDescriptor) columnFamily).build());
+    return FutureUtils
+        .toCompletableFuture(modifyColumnAsync(tableName, columnName, "add", modification))
+        .thenApply(r -> null);
   }
 
   @Override
-  public List<SnapshotDescription> listSnapshots()
-      throws IOException {
-    throw new IllegalArgumentException("listSnapshots is not supported");
+  public void deleteColumnFamily(TableName tableName, byte[] columnName) throws IOException {
+    deleteColumn(tableName, columnName);
   }
 
-
   @Override
-  public void deleteColumnFamily(TableName arg0, byte[] arg1) throws IOException {
-    throw new UnsupportedOperationException("deleteColumnFamily"); // TODO
+  public Future<Void> deleteColumnFamilyAsync(TableName tableName, byte[] columnName) 
+      throws IOException {
+    final String columnNameStr = Bytes.toString(columnName);
+    Modification.Builder modification = Modification.newBuilder().setId(columnNameStr)
+        .setDrop(true);
+    return FutureUtils.toCompletableFuture(modifyColumnAsync(
+        tableName, 
+        columnNameStr, 
+        "delete", 
+        modification)).thenApply(r -> null);
+  }
+
+
+  protected ListenableFuture<Table> modifyColumnAsync(TableName tableName, String columnName,
```

Comment:
Please have this return a `CompletableFuture<Void>` so that callers don't have to repeat the translation over and over.

Reply:
Make sense.